### PR TITLE
Fix solaris 11 upgrade 4.3

### LINKED ISSUE #1022: Unknown group id ossec in solaris 11 4.3 package generation
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.3.0 | Agent | Package generation | Solaris11 |

Hello team, when fixing the packages_builder_solaris pipeline in jenkins, I have encountered this error in the build process:

- https://ci.wazuh.info/job/Packages_builder_solaris/15337/

```
14:34:57      solaris11_cmake: chown: unknown group id ossec
14:34:57      solaris11_cmake: chown: unknown group id ossec
```

Specifically, the problem is in:

https://github.com/wazuh/wazuh-packages/blob/6b9b416e87a15cd16961ed1d725988a0293e25c4/solaris/solaris11/generate_wazuh_packages.sh#L175

https://github.com/wazuh/wazuh-packages/blob/6b9b416e87a15cd16961ed1d725988a0293e25c4/solaris/solaris11/generate_wazuh_packages.sh#L176

### PR DESCRIPTION
**Pull request by rauldpm**
|Related issue|
|---|
|closes #1022 |

## Changes

- Fixed postinstall.sh script run after package install.
- Removed unnecessary references to ossec group/user and files from generation script
- Fixed tabulations
- Improved help message
- Fixed documentation. Its necessary to copy the desired solaris directory into package_generation/src to generate the package using vagrant

## Logs example

- [solaris11.log](https://github.com/wazuh/wazuh-packages/files/7810506/sol11.log)

## Tests

<details><summary>Package install</summary>

```
root@solaris-vagrant:/vagrant# pkg install -g wazuh-agent_v4.3.0-sol11-i386.p5p wazuh-agent
           Packages to install:  1
            Services to change:  1
       Create boot environment: No
Create backup boot environment: No

DOWNLOAD                                PKGS         FILES    XFER (MB)   SPEED
Completed                                1/1         85/85      5.7/5.7  2.7M/s

PHASE                                          ITEMS
Installing new actions                       135/135
Updating package state database                 Done 
Updating package cache                           0/0 
Updating image state                            Done 
Creating fast lookup database                   Done 
Updating package cache                           2/2 
root@solaris-vagrant:/vagrant# /var/ossec/bin/wazuh-control start
Starting Wazuh v4.3.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
root@solaris-vagrant:/vagrant#
```

</details>

<details><summary>Package upgrade from 4.2.5 connected agent and check rids files</summary>

```
root@solaris-vagrant:/vagrant# pkg install -g wazuh-agent_v4.2.5-sol11-i386.p5p wazuh-agent
           Packages to install:  1
            Services to change:  1
       Create boot environment: No
Create backup boot environment: No

DOWNLOAD                                PKGS         FILES    XFER (MB)   SPEED
Completed                                1/1         86/86      5.5/5.5 10.8M/s

PHASE                                          ITEMS
Installing new actions                       131/131
Updating package state database                 Done 
Updating package cache                           0/0 
Updating image state                            Done 
Creating fast lookup database                   Done 
Updating package cache                           2/2 
root@solaris-vagrant:/vagrant# nano /var/ossec/etc/ossec.conf 
root@solaris-vagrant:/vagrant# /var/ossec/bin/wazuh-control start
Starting Wazuh v4.2.5...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
root@solaris-vagrant:/vagrant# ls -l /var/ossec/queue/rids/
total 4
-rw-r--r--   1 ossec    ossec          7 Jan  4 22:52 014
-rw-r--r--   1 ossec    ossec          5 Jan  4 22:52 sender_counter
root@solaris-vagrant:/vagrant# /var/ossec/bin/wazuh-control stop
Killing wazuh-modulesd... 
Killing wazuh-logcollector... 
Killing wazuh-syscheckd... 
Killing wazuh-agentd... 
Killing wazuh-execd... 
Wazuh v4.2.5 Stopped
root@solaris-vagrant:/vagrant# pkg install -g wazuh-agent_v4.3.0-sol11-i386.p5p wazuh-agent
            Packages to update:   1
            Services to change:   1
       Create boot environment:  No
Create backup boot environment: Yes

DOWNLOAD                                PKGS         FILES    XFER (MB)   SPEED
Completed                                1/1         74/74      5.7/5.7  9.4M/s

PHASE                                          ITEMS
Removing old actions                             9/9
Installing new actions                         11/11
Updating modified actions                    117/117
Updating package state database                 Done 
Updating package cache                           1/1 
Updating image state                            Done 
Creating fast lookup database                   Done 
Updating package cache                           2/2 
root@solaris-vagrant:/vagrant# ls -l /var/ossec/queue/rids/
total 4
-rw-r--r--   1 wazuh    wazuh          7 Jan  4 22:52 014
-rw-r--r--   1 wazuh    wazuh          5 Jan  4 22:53 sender_counter
root@solaris-vagrant:/vagrant# /var/ossec/bin/wazuh-control start
Starting Wazuh v4.3.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
root@solaris-vagrant:/vagrant#
```

</details>

---

- Build the package in any supported platform
  - [x] Solaris
- [x] Package installation
- [x] Package upgrade
- [x] Package remove
- [x] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [x] Build the package for i386
- Tests for Solaris
  - [x] Test the package on Solaris 11

### CODE REVIEW

File: solaris/solaris10/generate_wazuh_packages.sh [hunk 1]
```diff
@@ -295,17 +295,33 @@ build(){
...
-  exit $1
+    echo -e ""
+    echo -e "NAME"
+    echo -e "        $(basename $0) - Generate a Solaris 10 package"
+    echo -e ""
+    echo -e "SYNOPSIS"
+    echo -e "        $(basename $0) [OPTIONS]"
+    echo -e ""
+    echo -e "DESCRIPTION"
+    echo -e "        -b, --branch <branch>"
```
Comment (alberpilot) [inline solaris/solaris10/generate_wazuh_packages.sh:306]: Sort the options elements alphabetically

File: solaris/solaris11/generate_wazuh_packages.sh [hunk 2]
```diff
@@ -298,17 +283,33 @@ ctrl_c() {
...
-  exit $1
+    echo -e ""
+    echo -e "NAME"
+    echo -e "        $(basename $0) - Generate a Solaris 11 package"
+    echo -e ""
+    echo -e "SYNOPSIS"
+    echo -e "        $(basename $0) [OPTIONS]"
+    echo -e ""
+    echo -e "DESCRIPTION"
+    echo -e "        -b, --branch <branch>"
```
Comment (alberpilot) [inline solaris/solaris11/generate_wazuh_packages.sh:294]: Same

File: solaris/solaris11/postinstall.sh [hunk 3]
```diff
@@ -1,16 +1,17 @@
 #!/bin/sh
 # postinst script for wazuh-agent
-# Wazuh, Inc 2015-2020
+# Wazuh, Inc 2015-2021
```
Comment (alberpilot) [inline solaris/solaris11/postinstall.sh:3]: 2022. Please replace all years by this new one.

File: solaris/solaris11/postinstall.sh [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
 
 install_path="<INSTALL_PATH>"
 osversion=$(uname -v)
 SCA_BASE_DIR="${install_path}/ruleset/sca/"
 SCA_TMP_DIR="${install_path}/tmp/sca"
 
+# upgrade from 3
 if [ -d ${install_path}/logs/ossec ]; then
-  if [ -z "$(ls -A ${install_path}/logs/ossec)" ]; then
+  if [ -z "$(ls -A ${install_path}/logs/ossec)" ]; then  # empty
```
Comment (alberpilot) [inline solaris/solaris11/postinstall.sh:12]: Do you consider necessary the comment? `-z` is known that returns true if the value is empty.

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

**Review by verdx — APPROVED** [review]
LGTM

File: solaris/solaris10/generate_wazuh_packages.sh (hunk 1 above)
Comment (rauldpm) [reply solaris/solaris10/generate_wazuh_packages.sh:306]: Fixed in 51c6056cd937c0017172a0571a404418f91c9bb2

File: solaris/solaris11/generate_wazuh_packages.sh (hunk 2 above)
Comment (rauldpm) [reply solaris/solaris11/generate_wazuh_packages.sh:294]: Fixed in 51c6056cd937c0017172a0571a404418f91c9bb2

File: solaris/solaris11/postinstall.sh (hunk 3 above)
Comment (rauldpm) [reply solaris/solaris11/postinstall.sh]: Fixed in aa77fe7e88c124e77fbe9aae8edb13afab09a472 and 869642c979b33d843d77f2de525b94fa1ceb5bf0

File: solaris/solaris11/postinstall.sh (hunk 4 above)
Comment (rauldpm) [reply solaris/solaris11/postinstall.sh]: I do not remember adding these comments, but it seems that I put them at some point. Removed in acc587f0aa60ee65e0d73900caede7e7237055aa

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**